core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -267,6 +281,18 @@ class Partition(val topicPartition: TopicPartition,
     leaderLogIfLocal.exists { inSyncReplicaIds.size == _.config.minInSyncReplicas }
   }
 
+  def isReassigning: Boolean = assignmentState.isInstanceOf[OngoingReassignmentState]
+
+  def isAddingLocalReplica: Boolean = assignmentState match {
+    case state: OngoingReassignmentState => state.addingReplicas.contains(localBrokerId)
```

Comment:
This is another case where we could move the logic into `AssignmentState`.

Reply:
Thanks, I'll move it there